debezium-embedded/src/test/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotTest.java

```diff
@@ -386,6 +391,31 @@ public void snapshotOnlyWithRestart() throws Exception {
         }
     }
 
+    @Test
+    @FixFor("DBZ-4272")
+    public void snapshotProceededBySchemaChange() throws Exception {
+        Testing.Print.enable();
+
+        populateTable();
+        startConnector();
+        waitForConnectorToStart();
+
+        // Initiate a schema change to the table immediately before the adhoc-snapshot
+        try (JdbcConnection connection = databaseConnection()) {
+            connection.execute(alterTableStatement(tableName()));
+        }
+
+        // Some connectors, such as PostgreSQL won't be notified of the previous schema change
+        // until a DML event occurs, but regardless the incremental snapshot should succeed.
+        sendAdHocSnapshotSignal();
```

Comment:
It would be good to also test that snapshot works when a column is dropped. I believe this was failing as well, but in a different place